src/auth/baseexternalclient.ts

```diff
@@ -489,4 +503,32 @@ export abstract class BaseExternalAccountClient extends AuthClient {
       return this.scopes;
     }
   }
+
+  /**
+   * Checks whether Google APIs URL is valid.
+   * @param prefix The prefix of url.
+   * @param url The url from external account options object.
+   * @return True if it is null or can match to one of the valid token urls.
+   *   Otherwise, returns false.
+   */
+  private validateGoogleAPIsUrl(prefix: string, url?: string): boolean {
+    if (typeof url === 'undefined') {
+      return true;
+    }
+    const matchUrlPatterns: string[] = [];
+    matchUrlPatterns.push(
```

Comment:
wondering if this could just be defined as a `const` at the top of the file, rather than us using `push` to populate an array/